### PR TITLE
[IMP] open_academy: relation between models t#59945

### PR DESCRIPTION
**Pull request by JJRR5**
**Finished the Second Section of Building a Module Course**
The main goal of this section was to understand the importance and how the relational fields works

- Using a many2many field created 3 fields : responsible_user_id , instructor_id and courses_id.
- Using a one2many field created a field "session_id" which main goal is to reflect the relation between courses and models
- Using a many2many field created a field which main goal is to relate every session to a set of attendees

**Image 1.  Course View with their respective relational fields**

image

**Image 2. Sessions View with their respective relational fields**

image

### CODE REVIEW

**Comment by JJRR5** [comment]
@sebasdrk17 Good morning Alexis, could you please check this PR please?

**Comment by sebasdrk17** [comment]
I'ts not an [ADD] anymore, the title should be [IMP]

File: open_academy/models/course.py [hunk 1]
```diff
@@ -8,3 +8,5 @@ class Course(models.Model):
     name = fields.Char(required=True)
     title = fields.Text()
     description = fields.Text()
+    responsible_user_id = fields.Many2one('res.users')
+    session_id = fields.One2many('session', 'courses_id')
```
Comment (sebasdrk17) [inline open_academy/models/course.py]: This field is one2many, you need to set the `_ids` so we know that that field will have multiple records

```suggestion
    session_ids = fields.One2many('session', 'courses_id')
```

File: open_academy/models/session.py [hunk 2]
```diff
@@ -7,3 +7,6 @@ class Session(models.Model):
     start_date = fields.Date()
     duration = fields.Integer()
     number_of_seats = fields.Integer()
+    instructor_id = fields.Many2one('res.partner')
+    courses_id = fields.Many2one('course', required=True)
+    attendees_id = fields.Many2many('res.partner', string="Attendees")
```
Comment (sebasdrk17) [inline open_academy/models/session.py]:
```suggestion
    attendees_ids = fields.Many2many('res.partner', string="Attendees")
```

**Review by sebasdrk17 — CHANGES_REQUESTED** [review]
Asked some changes

**Comment by JJRR5** [comment]
Changes done, @sebasdrk17 could you please check?

File: open_academy/views/course_views.xml [hunk 3]
```diff
@@ -9,6 +9,8 @@
                     <group>
                         <group>
                             <field name="name"/>
+                            <field name="responsible_user_id"/>
+                            <field name="session_id"/>
```
Comment (sebasdrk17) [inline open_academy/views/course_views.xml]: You changed this field from `session_id` to `session_ids`, please whenever you get a review test again all your changes

We do not want a server error in production

**Review by sebasdrk17 — CHANGES_REQUESTED** [review]
Test your changes again and ensure everything is working please

File: open_academy/views/course_views.xml (hunk 3 above)
Comment (JJRR5) [reply open_academy/views/course_views.xml]: Take it for grantend

**Comment by JJRR5** [comment]
I tested my changes, and everything it's ok, could you please check? @sebasdrk17

File: open_academy/models/session.py [hunk 4]
```diff
@@ -7,3 +7,6 @@ class Session(models.Model):
     start_date = fields.Date()
     duration = fields.Integer()
     number_of_seats = fields.Integer()
+    instructor_id = fields.Many2one('res.partner')
+    courses_id = fields.Many2one('course', required=True)
```
Comment (sebasdrk17) [inline open_academy/models/session.py]: I missed this field, a session can only have one course, so it should be singular

```suggestion
    course_id = fields.Many2one('course', required=True)
```

**Review by sebasdrk17 — CHANGES_REQUESTED** [review]
Change the field and when you push please make a squash

**Comment by JJRR5** [comment]
@sebasdrk17 squash done, could you please check?

**Review by sebasdrk17 — APPROVED** [review]
@andreaarce could you check please?

**Review by andreaarce — APPROVED** [review]

**Comment by andreaarce** [comment]
LGTM

**Pull request merged**